dfusion_rust_core/src/models/order.rs

```diff
@@ -47,6 +47,73 @@ impl Order {
             sell_amount,
         }
     }
+
+    /// Calculates the utility of an order based on a buy and sell price as well
+    /// as the executed buy amount.
+    ///
+    /// # Returns
+    ///
+    /// Returns utility of an order or None when there was an error in the
+    /// calculation such as overflow or divide by 0.
+    pub fn utility(
+        &self,
+        buy_price: u128,
+        exec_buy_amount: u128,
+        exec_sell_amount: u128,
+    ) -> Option<U256> {
+        let buy_price = u128_to_u256(buy_price);
+        let exec_buy_amount = u128_to_u256(exec_buy_amount);
+        let exec_sell_amount = u128_to_u256(exec_sell_amount);
+        let buy_amount = u128_to_u256(self.buy_amount);
+        let sell_amount = u128_to_u256(self.sell_amount);
+
+        // essential_utility = (exec_buy_amount - (exec_sell_amount * buy_amount)
+        //   / sell_amount) * buy_price
+        // essential_utility = (((exec_sell_amount * buy_amount) % sell_amount)
+        //   * buy_price) / sell_amount
+        // utility = essential_utility - essential_utility
+
+        let essential_utility = exec_buy_amount
+            .checked_sub((exec_sell_amount * buy_amount).checked_div(sell_amount)?)?
+            * buy_price;
+        let utility_error =
+            (((exec_sell_amount * buy_amount) % sell_amount) * buy_price) / sell_amount;
+        let utility = essential_utility.checked_sub(utility_error)?;
+
+        Some(utility)
+    }
+
+    /// Calculates the disregarded utility of an order based on a buy and sell
+    /// price as well as the executed sell amount.
+    ///
+    /// # Returns
+    ///
+    /// Returns disregarded utility of an order or None when there was an error
+    /// in the calculation such as overflow or divide by 0.
+    pub fn disregarded_utility(
+        &self,
+        buy_price: u128,
+        sell_price: u128,
+        exec_sell_amount: u128,
+    ) -> Option<U256> {
+        let buy_price = u128_to_u256(buy_price);
+        let sell_price = u128_to_u256(sell_price);
+        let exec_sell_amount = u128_to_u256(exec_sell_amount);
+        let buy_amount = u128_to_u256(self.buy_amount);
+        let sell_amount = u128_to_u256(self.sell_amount);
+
+        // limit_term = sell_price * sell_amount - buy_amount * buy_price
```

Comment:
Do you think this comment is necessary?

Reply:
I had used them as a reference when I was writing the calculations. I guess these comments can be removed since they add no value: they are basically an almost verbatim copy of the code below them. They just confuse people when they are wrong like :point_up:.